src/common/function/src/scalars/matches.rs

```diff
@@ -0,0 +1,802 @@
+            vec![
+                ConcreteDataType::string_datatype(),
+                ConcreteDataType::string_datatype(),
+            ],
+            Volatility::Immutable,
+        )
+    }
+
+    // TODO: read case-sensitive config
+    fn eval(&self, _func_ctx: FunctionContext, columns: &[VectorRef]) -> Result<VectorRef> {
+        ensure!(
+            columns.len() == 2,
+            InvalidFuncArgsSnafu {
+                err_msg: format!(
+                    "The length of the args is not correct, expect exactly 2, have: {}",
+                    columns.len()
+                ),
+            }
+        );
+        ensure!(
+            columns[1].len() == 1,
+            InvalidFuncArgsSnafu {
+                err_msg: "The second argument should be a string literal",
+            }
+        );
+        let pattern_vector = &columns[1]
+            .cast(&ConcreteDataType::string_datatype())
+            .context(InvalidInputTypeSnafu {
+                err_msg: "cannot cast `pattern` to string",
+            })?;
+        // Safety: both length and type are checked before
+        let pattern = pattern_vector.get(0).as_string().unwrap();
+        self.eval(columns[0].clone(), pattern)
+    }
+}
+
+impl MatchesFunction {
+    fn eval(&self, data: VectorRef, pattern: String) -> Result<VectorRef> {
+        let col_name = "data";
+        let parser_context = ParserContext::default();
+        let ast = parser_context.parse_pattern(&pattern)?;
+        let like_expr = ast.to_like_expr(col_name);
+
+        let input_schema = Self::input_schema();
+        let session_state =
+            SessionState::new_with_config_rt(SessionConfig::default(), Arc::default());
+        let planner = DefaultPhysicalPlanner::default();
+        let physical_expr = planner
+            .create_physical_expr(&like_expr, &input_schema, &session_state)
+            .context(GeneralDataFusionSnafu)?;
+
+        let data_array = data.to_arrow_array();
+        let arrow_schema = Arc::new(input_schema.as_arrow().clone());
+        let input_record_batch = RecordBatch::try_new(arrow_schema, vec![data_array]).unwrap();
+
+        let num_rows = input_record_batch.num_rows();
+        let result = physical_expr
+            .evaluate(&input_record_batch)
+            .context(GeneralDataFusionSnafu)?;
+        let result_array = result
+            .into_array(num_rows)
+            .context(GeneralDataFusionSnafu)?;
+        let result_vector =
+            BooleanVector::try_from_arrow_array(result_array).context(IntoVectorSnafu {
+                data_type: DataType::Boolean,
+            })?;
+
+        Ok(Arc::new(result_vector))
+    }
+
+    fn input_schema() -> DFSchema {
+        DFSchema::from_unqualifed_fields(
+            [Arc::new(Field::new("data", DataType::Utf8, true))].into(),
+            HashMap::new(),
+        )
+        .unwrap()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum PatternAst {
```

Comment:
Great work! I believe we can divide this file into a separate module such as `match` or similar. The aim is to keep this file concise and organized. Take your time, it's just a suggestion.